src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,111 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int CAPACITY = 16;
+    private static final int CAPACITY_EXPAND = 8;
+    private static final float LOAD_FACTOR = 0.75F;
+    private Node<K, V>[] bucket = new Node[CAPACITY];
+    private int size;
+
+    private static class Node<K, V> {
+        private int hash;
+        private final K key;
+        private V value;
+        private Node<K, V> next;
+
+        public Node(int hash, K key, V value, Node<K, V> next) {
+            this.hash = hash;
+            this.key = key;
+            this.value = value;
+            this.next = next;
+        }
+    }
+
+    private int indexCalculation(K key) {
+        if (key != null) {
+            if (key.hashCode() < 0) {
+                return key.hashCode() % CAPACITY + CAPACITY_EXPAND;
+            }
+            return key.hashCode() % CAPACITY;
+        } else {
+            return 0;
+        }
+    }
 
     @Override
     public void put(K key, V value) {
-
+        int arraySize = indexCalculation(key);
+        Node<K, V> tempNode;
+        Node<K, V> newNode = new Node<>(key != null ? key.hashCode() : 0, key, value, null);
+        tempNode = bucket[arraySize];
+        while (true) {
```

Comment:
Do not use `while (true)`. Using` break;` is also a bad practice.
Use a loop for going along the chain **after** the check that tempNode != null